### DISCUSSION TITLE
@remotion/paths use in Node-RED

### DISCUSSION THREAD

**Discussion by MadTooler** [remotion-dev/remotion#4378] [Q&A] [answered]
Hello all!

I have been looking at svg path editing options to use within the server side of Node-RED's function environment.  For now, I want to keep away from in browser and DOM limited approaches.  CLI could be fine, but js modules are preferred.

One of the more important modules I am interested in apply is warp.js.  In researching that module, I found remotion has coupled and updated warp with a handful of other modules I am also working to apply!  What I would like to find out is can the @remotion/paths will work without browser?

Thanks for any advice

**Comment by JonnyBurger** [comment, marked as answer]
Yes, `@remotion/paths` works outside of a browser!

**Reply by MadTooler** [reply]
Excellent!

Anyone try it with Node-RED?  I will give it a try and see what I can break.

**Reply by MadTooler** [reply]
@JonnyBurger 

It does runs in Node_RED!  Super Cool!

I have tested a handful of the features with working results as with your examples.

The next task is trying to replicate what I saw on CodePen doing something like an envelope distort/warp of custom paths (not text and not in browser with all the DOM stuff).  I am a little mixed up on how to translate the syntax of the warp.js original calls with what remotion wants.  To dumb it down with a sprinkle of technical error, warp.js allowed for additional parameters to be created at an initial creation of the warp function with a later call that then made the warp.  See [here.](https://codepen.io/figle/pen/BZvyQZ)

Is this still possible with your reworking of the module, or has there been some loss?

Also, I am not clear on how to pass in the interpolate value if I want to change it from the 1% default.  What is the appropriate method there?

**Reply by MadTooler** [reply]
After a bit more digging and testing, is it correct that the remotion warp function is used for single xy coord modification only?

The warp.js module seems to send an array of all the xy coordinates to the "transform" function which allows the envelope example I noted to both modify the coords as well as work out the angles and weights of the path based upon the following coordinates.  If the remixed remotion version doesn't have the ability to receive more at once (I am not sure if there is an option for this or not), I don't think I can get the intended envelope effect as with the example on CodePen.

What are your thoughts?

**Reply by JonnyBurger** [reply]
Yes the idea was never to have the same feature set as warpjs

You can also copy paste the source code of our functions if you need to make modifications to it

**Reply by MadTooler** [reply]
Thanks for the reply. 

I asked enough questions I confused myself with your answer.

To clarify, can your module perform the kind of envelope distort as warpjs?
